docs/source/workflows/data_sources.rst

```diff
@@ -0,0 +1,94 @@
+Data Sources
+=============
+
+Data sources are used by modules to pull data from outside of the AI engine.
+For example, a :doc:`predictor <predictors>` may need to define external training data.
+
+CSV Data Source
+----------------
+
+The :class:`~citrine.informatics.data_sources.CSVDataSource` draws data from a CSV file stored on the data platform and annotates it by mapping each column name to a :class:`~citrine.informatics.descriptors.Descriptor`.
+The file is referenced via a :class:`~citrine.resources.file_link.FileLink`.
+Each FileLink references an explicit version of the CSV file.
+Uploading a new file with the same name will produce a new version of the file with a new FileLink.
+
+The columns in the CSV are extracted and parsed by a mapping of column header names to user-created descriptors.
+Columns in the CSV that are not mapped with a descriptor are ignored.
+
+Assume that a file data.csv exists with the following contents:
```

Comment:
@floparedes @maxhutch is there a limit on the number of rows/columns/cells in a CSV data source? I see limits on Ara tables here and total number of training data for a predictor, but am not sure which parts transfer to CSV data source: https://citrine.atlassian.net/wiki/spaces/PLA/pages/251297805/AI+Engine+limitations+and+questions+for+CP+Onboarding

Reply:
There is not currently a limit.  I think we should cap the file size because we read the whole file into memory in orion.  cc: @andyczerwonka @jtharris